### PR TITLE
Invalidate product index after change product position in the category

### LINKED ISSUE #28149: Changing product position in the category has no effect on the front-end
### Preconditions (*)
1. Magento 2.4-develop
2 All indexes mode is "update by schedule"
Capture 2020-05-08 at 9 25 50

### Steps to reproduce (*)
1. Go to admin panel -> catalog -> inventory -> categories
2. Click on any category with the products and visible on the front-end
3. On the "Products in Category" tab change position of some product (do not delete or add any product from/to category)
4. click the "Save" button

### Expected result (*)

1. Product indexes are invalidated
2. After running cron jobs "indexer_reindex_all_invalid" and "indexer_update_all_views" product position on the category page is changed on the frontend.

### Actual result (*)

1. All indexes are still valid
2. The product position on the category page on the frontend is not changed.

### Additional information

Detailed steps and examples defined during confirmation: https://github.com/magento/magento2/issues/28149#issuecomment-629136334

### PR DESCRIPTION
**Pull request by aleksinoleg**
### Description (*)

Dispatching event "catalog_category_change_products" if there are products with updated position while saving the category

### Fixed Issues (if relevant)

1. Fixes magento/magento2#28149: Changing product position in the category has no effect on the front-end

### Manual testing scenarios (*)

1. Go to admin panel -> catalog -> inventory -> categories
2. Click on any category with the products and visible on the front-end
3. On the "Products in Category" tab change position of some product (do not delete or add any product from/to category)
4. Click the "Save" button

**Expected result**
1. Product indexes are invalidated
2. After running cron jobs "indexer_reindex_all_invalid" and "indexer_update_all_views" product position on the category page is changed on the frontend.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
Hi @aleksinoleg,
Could you sign Adobe CLA and cover this case with MFTF test?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @aleksinoleg,
Could you sign Adobe CLA and cover this case with MFTF test?

**Comment by aleksinoleg** [comment]
Will reopen it again

**Comment by ihor-sviziev** [comment]
Just requesting my changes again in order to have correct PR status

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @aleksinoleg,
Could you sign Adobe CLA and cover this case with MFTF test?

**Comment by aleksinoleg** [comment]
I have no ability to cover these changes by tests. Could somebody help me with that?

File: app/code/Magento/Catalog/Model/ResourceModel/Category.php [hunk 1]
```diff
@@ -1078,7 +1078,6 @@ public function countVisible()
      */
     public function load($object, $entityId, $attributes = [])
     {
-        $this->_attributes = [];
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Model/ResourceModel/Category.php:1081]: This is fix for static test failure that property was declared dynamically and actually never used.

**Comment by ihor-sviziev** [comment]
@engcom-Charlie could you help with covering this change with mftf test?

**Comment by engcom-Charlie** [comment]
Hi @ihor-sviziev, yes sure.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
The issue was only in the case indexer `catalog_category_product` => `UPDATE BY SCHEDULE`
We change product position on a category page but indexer was still valid(incorrect).
But if we run reindex as result we have correct sorting on front.
So I think that checking indexer status after changing positions will be enough.

@ihor-sviziev, What do You think?

**Comment by ihor-sviziev** [comment]
@engcom-Charlie looks good. Test were failing on 2.4-develop branch ?

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7554](https://jira.corp.magento.com/browse/ENGCOM-7554) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@aleksinoleg thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Create a category with a few simple products;
2. Set all indexes mode to "Update by Schedule" (Admin->System->Index Management);
Screenshot from 2020-05-15 12-33-13
3. Go to Admin -> Catalog -> Inventory -> Categories;
4. Click on any category with the products and visible on the front-end;
5. On the "Products in Category" tab change position of some product (do not delete or add any product from/to category)
screenshot_11
6. Go to Storefront -> Category with products;

**Before:**

:heavy_multiplication_x:  All indexes are still valid

Screenshot from 2020-05-15 12-30-05

:heavy_multiplication_x: The product position on the category page on the frontend is not changed.

Screenshot from 2020-05-15 12-24-26

**After:**

:heavy_check_mark:  Product indexes are invalidated;

Screenshot from 2020-05-15 13-21-21

:heavy_check_mark:  After running cron jobs "indexer_reindex_all_invalid" and "indexer_update_all_views" product position on the category page is changed on the frontend.

Screenshot from 2020-05-15 13-37-17

**Pull request merged**